arch/mips/core/thread.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2017 Imagination Technologies Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <kernel_structs.h>
+#include <wait_q.h>
+#include <string.h>
+#include <soc.h>
+
+void _thread_entry_wrapper(k_thread_entry_t thread,
+			   void *arg1,
+			   void *arg2,
+			   void *arg3);
+
+void _new_thread(struct k_thread *thread, k_thread_stack_t stack,
+		 size_t stack_size, k_thread_entry_t entry,
+		 void *arg1, void *arg2, void *arg3,
+		 int prio, unsigned int options)
+{
+	_ASSERT_VALID_PRIO(prio, entry);
+
+	char *stack_memory = K_THREAD_STACK_BUFFER(stack);
+	char *ctx = (char *)STACK_ROUND_DOWN(stack + stack_size -
```

Comment:
This is wrong, please initialize ctx in terms of stack_memory not stack.
k_thread_stack_t is an opaque type, used for inserting guard regions for memory protection scenarios.